prov/shm/src/smr_ep.c

```diff
@@ -1354,6 +1355,11 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 		else
 			ep->smr_progress_ipc_list = smr_progress_ipc_list_noop;
 
+		if (smr_env.use_xpmem && xpmem)
```

Comment:
maybe I'm missing something. what is the xpmem variable here?

Reply:
It's a global xpmem variable which holds information about the exported memory address space of the process which is exported via XPMEM. It will be initialized if XPMEM is available. but you're right, it's redundant to explicitly check it here.